src/building/barracks.c

```diff
@@ -79,6 +83,29 @@ static int get_closest_legion_needing_soldiers(const building *barracks)
     int recruit_type = LEGION_RECRUIT_NONE;
     int min_formation_id = 0;
     int min_distance = INFINITE;
+
+    int priority_recruitment = barracks->subtype.barracks_priority;
+    int requiered_recruitment = recruit_type;
```

Comment:
```suggestion
    int required_recruitment = recruit_type;
```

Don't forget to change the variable name elsewhere.